src/main/java/org/commcare/core/graph/c3/AxisConfiguration.java

```diff
@@ -162,11 +158,19 @@ private void addTickConfig(JSONObject axis, String key, String varName) throws G
             }
         }
 
+        if ((isX && !isRotatedBarGraph) || (isRotatedBarGraph && key.startsWith("y"))) {
+            String largestLabel = mData.getLargestXLabel();
+            if (largestLabel != null && !largestLabel.isEmpty()) {
+                int largestHeight = StringWidthUtil.getStringWidth(largestLabel);
+                if (height < largestHeight) {

```

Comment:
puzzled why do we need to redo the comparison here when we are already using the `largestLabel` to get the height.

Reply:
Here I'm using `mData.getLargestXLabel()` which is getting the value from `data` configuration. It is different than the values used above in the json config.

Reply:
I see, it would be nice to rename the variables here accordingly. ` if (height < largestHeight) {` the statement is confusing because semantically it's always true.